src/index.js

```diff
@@ -30,6 +31,48 @@ newTodoInputEl.getElement().addEventListener("keypress", (e) => {
   }
 });
 
+const handlePendingTaskStatus = (e, taskItemEl) => {
+  if (
+    e.target.tagName === "INPUT" &&
+    e.target.getAttribute("type") === "checkbox"
+  ) {
+    if (e.target.checked) {
+      e.target.setAttribute("checked", "");
+      taskItemEl.setAttribute("class", "completed");
+    } else {
+      e.target.removeAttribute("checked");
+      taskItemEl.removeAttribute("class");
+    }
+  }
+};
+
+let mouseClicks = 0;
+let timer = null;
+
+const handleEditTaskMode = (e, taskItemEl) => {
+  if (e.target.tagName === "LABEL") {
+    const checkboxEl = e.target.previousSibling.previousSibling;
+    mouseClicks++;
+    if (mouseClicks === 1) {
+      timer = setTimeout(() => {
+        mouseClicks = 0;
+      }, 500);
+    }
+    if (mouseClicks === 2 && !checkboxEl.checked) {
```

Comment:
handle a double click event (it should be twice clicked register before 500 milliseconds otherwise it won't be considered as double click) to enable the edit input field element.